testing/unittests/unittests.cpp

```diff
@@ -569,5 +570,71 @@ class StringBufferTest : public CppUnit::TestFixture
 CPPUNIT_TEST_SUITE_REGISTRATION( StringBufferTest );
 CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( StringBufferTest, "StringBufferTest" );
 
+StringBuffer &mbToBase58(StringBuffer &s, const MemoryBuffer &data)
+{
+    size_t b58Length = data.length() * 2 + 1;
+
+    if (b58enc(s.reserve(b58Length), &b58Length, data.toByteArray(), data.length()) && b58Length > 1)
+        s.setLength(b58Length);
+    return s;
+}
+
+StringBuffer &base64ToBase58(StringBuffer &s, const char *b64)
+{
+    MemoryBuffer mb;
+    JBASE64_Decode(b64, mb);
+    return mbToBase58(s, mb);
+}
+
+StringBuffer &textToBase58(StringBuffer &s, const char *text)
+{
+    MemoryBuffer mb;
+    mb.append((size_t)strlen(text), text);
+    return mbToBase58(s, mb);
+}
+
+class Base58Test : public CppUnit::TestFixture
+{
+    CPPUNIT_TEST_SUITE( Base58Test  );
+        CPPUNIT_TEST(testEncode);
+    CPPUNIT_TEST_SUITE_END();
+
+    void testEncode()
+    {
+        StringBuffer s;
+
+        //short string
+        ASSERT(streq(textToBase58(s.clear(), "1"), "r"));
+        ASSERT(streq(textToBase58(s.clear(), "\xFF"), "5Q"));
+
+        //text string
+        textToBase58(s.clear(), "Fifty-eight is the sum of the first seven prime numbers.");
+        ASSERT(streq(s, "2ubdTkzo5vaWL4FKQGro88zp8v6Q5EftVBq2fbZsWCDRzQxGDb1heKFsMReJNhsRsK6TfvrgqVeRB"));
+
+        //hex 005A1FC5DD9E6F03819FCA94A2D89669469667F9A074655946
+        base64ToBase58(s.clear(), "AFofxd2ebwOBn8qUotiWaUaWZ/mgdGVZRg==");
+        ASSERT(streq(s, "19DXstMaV43WpYg4ceREiiTv2UntmoiA9j"));
+
+        base64ToBase58(s.clear(), "/u+u8CL6");
+        ASSERT(streq(s, "3Bx9Y4pUR"));
+
+        //hex FFFEEFAEF022FA
+        base64ToBase58(s.clear(), "//7vrvAi+g==");
+        ASSERT(streq(s, "AhfV5sjWb3"));
+
+        //This input causes the loop iteration counter to go negative
+        //hex 00CEF022FA
+        base64ToBase58(s.clear(), "AM7wIvo=");
+        ASSERT(streq(s, "16Ho7Hs"));
+
+        //empty input
+        MemoryBuffer mb;
+        size_t b58Length = 0;
+        ASSERT(!b58enc(s.reserve(b58Length), &b58Length, mb.toByteArray(), mb.length()));
```

Comment:
Is it correct that this fails?  When I commented that it returned false I wasn't convinced that was the correct result.

Reply:
I didn't think worth changing the 3rd party code as unlikely to be used this way, but I can take a look.